### DISCUSSION TITLE
Documentation of python.irobot.com and irobot_edu_sdk?

### DISCUSSION THREAD

**Discussion by framlin** [iRobotEducation/create3_docs#268] [Questions] [answered]
Hello, first of all, create3 is a great learning and experimentation tool.
Saying that, I am wondering, if there is a detailed documentation of the python.irobot.com page and the irobot_edu_sdk?

Something, that explains, what is exactly going on, if I run one of the examples? Where does the python scripts run? What happens? What is the functionality/API of the SDK? Can I "destroy" the firmware of the create3 and so on and so forth?

**Comment by shamlian** [comment, marked as answer]
Hi there,
Thanks! We're doing our best. :)

When you run code in the [iRobot® Python Web Playground](https://python.irobot.com) or using the `irobot_edu_sdk` iRobot® Education Python® SDK (offline version [now available here](https://github.com/iRobotEducation/irobot-edu-python-sdk/)), you are interacting with the robot using [Web Bluetooth](https://chromestatus.com/feature/5264933985976320) using the [Root Robot Bluetooth Low Energy Protocol](https://github.com/iRobotEducation/root-robot-ble-protocol). You are effectively teleoperating the robot over BLE. The robot translates between the BLE protocol and ROS 2 messages onboard. There is no danger of destroying anything on the Create 3 robot.

The SDK follows a similar syntax as Learning Level 3 of the [iRobot® Coding App](https://code.irobot.com). You can use the coding app as a cheat sheet, or else clicking the "Docs" button on the right hand side in the iRobot® Python Web Playground will get you some detail on most of the commands.

Hope this helps!

**Reply by azbitak-invn** [reply]
Hello,

What is the difference between web bluetooth and the common one? It seems the bluetooth of the robot is only visbile from chrome web playground, but couldn't see if from usual bluetooth sniffers. Is it a manipulation to do to see robot bluetooth from any where ?

Thanks

**Reply by shamlian** [reply]
iRobot Education's robots communicate using a common Bluetooth Low Energy serial protocol, the details of which can be found [here](https://github.com/iRobotEducation/root-robot-ble-protocol). The web browser uses a `web-bluetooth` backend while the native OS uses a Python cross-platform backend called [`bleak`](https://github.com/hbldh/bleak). Otherwise, they are nearly the same implementations of [iRobot Coding Level 3](code.robot.com).

**Comment by framlin** [comment]
Ahhh, ic.
Thank you very much.
